internal/config/config.go

```diff
@@ -37,6 +37,9 @@ var (
 	callbackLock sync.Mutex
 )
 
+// Config is the config for the fleet manager and agent. Each use slightly
+// different fields from this struct. It is stored as JSON in configmaps under
```

Comment:
Since we discussed it, I was about to try this yesterday, but then I vaguely remembered I tried it before and there was some shared code I didn't want to duplicate. Can't find that attempt, but better to do it in a separate PR :)

Reply:
Oh of course, by "eventually" I meant "not now", since it's understandably quite beyond the scope of this PR.